data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -582,6 +586,16 @@ private void logFailureForBulkRequests(final List<FailedBulkOperation> failedBul
   }
 
   private void logFailureForDlqObjects(final List<DlqObject> dlqObjects, final Throwable failure) {
+    if (getFailurePipeline() != null) {
+      List<Record<Event>> records = new ArrayList<>();
+      for (DlqObject dlqObject : dlqObjects) {
+        if (dlqObject.getEvent() != null) {
+          records.add(new Record<>(dlqObject.getEvent()));
+        }
+      }
+      getFailurePipeline().sendEvents(records);
```

Comment:
There is a big item here that we should consider. You are putting a `DlqObject` into the failure pipeline. But, this is not the original `Event`. So this means that we will have different data structures in failure pipelines for each sink and processor.

I tend to think that the best approach would be to include the original `Event` in the failure pipeline. Additionally, we should include failure information as metadata in that `Event`.

This also opens the question of whether we want to save `DlqObject` into the final sinks in the failure pipeline.